ros_docker_controller/docker_controller.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/python
+
+import os
+import shutil
+import docker
+import curses
+from cursesmenu import SelectionMenu
+import distutils.spawn
+import subprocess
+
+__author__ = 'Daniel Volya (RustyBamboo)'
+
+
+class DockerController(object):
+    def __init__(self, stdscr):
+        self.stdscr = stdscr
+
+        self.docker_client = docker.from_env()
+
+        # Check if docker server is running
+        try:
+            self.docker_client.info()
+        except Exception:
+            assert False, 'Is Docker server running?'
+
+        # Attempt to load container, if can't then later ask user to select image
+        self.container = None
+        # If container is currently running
+        if "RustyROS" in [
+                x.name for x in self.docker_client.containers.list()
+        ]:
+            self.container = self.docker_client.containers.get('RustyROS')
+        # Check if container is stopped, but still exists
+        else:
+            try:
+                self.container = self.docker_client.containers.get('RustyROS')
+                self.container.start()
+            except Exception:
+                self.container = None
+
+        # if no docker images are avaliable, go ahead and build one
+        if not self.docker_client.images.list():
+            self.brand_new_docker_image()
+
+        # attempt to find what kind of terminal emulator is being used
+        self.terminal = None
+        self.find_avaliable_terminal()
+        assert self.terminal is not None, 'No terminal found'
+
+        # run curses stuff
+        self.draw()
+
+    # Finds terminal emulator
+    def find_avaliable_terminal(self):
```

Comment:
I'm fairly confident there is a better way to do this, like asking the user which terminal they prefer then remembering a default. Just launching a terminal seems... arbitrary.

Reply:
Nah fam